PathfinderAPI/Util/XML/ElementInfo.cs

```diff
@@ -16,6 +19,20 @@ public class ElementInfo
     public Dictionary<string, string> Attributes { get; set; } = new Dictionary<string, string>();
     public List<ElementInfo> Children { get; set; } = new List<ElementInfo>();
     public ulong NodeID { get; } = freeId++;
+    public XmlNodeType Type { get; set; }
+    public bool IsText => Type == XmlNodeType.Text;
```

Comment:
This entire structure is confusing, why can a text element have attributes and children if they will never be serialized? This also makes adding text to an element two operations for a developer, since I have to both create a child element and also append to Content myself to keep everything up to date. These being out of sync really shouldn't be allowed.

Reply:
Backwards compatibility, this was addressed with [here](https://github.com/Arkhist/Hacknet-Pathfinder/pull/161#issuecomment-1110848934). It obviously doesn't make sense but without a refactor which would introduce breaking changes, which is addressed with #163, there is no safe way for mods to rely on the API's semver being true.